src/ring_theory/discrete_valuation_ring.lean

```diff
@@ -367,6 +379,96 @@ begin
   { apply (hirr.ne_zero (pow_eq_zero h)).elim, }
 end
 
+/-!
+## The additive valuation on a DVR
+-/
+
+/-- The `ℕ`-valued additive valuation on a DVR (returns junk at `0` rather than `+∞`) -/
+noncomputable def add_val (R : Type u) [integral_domain R] [discrete_valuation_ring R] : R → ℕ :=
```

Comment:
There's not much point, but you could declare this a `zero_hom` to get a tiny bit of bundling

Reply:
I almost sent 0 to 37. This definition is really an auxiliary one; I will only be using it when defining the correct object, which is either the map to `with_top nat` sending 0 to top, or (more likely) the map to `with_zero (multiplicative nat)` or `with_zero (multiplicative int)`, because the latter is the `valuation` right now (and the former isn't, but will be once I get round to generalising the definition of `valuation`).

Should I rename this function `add_val_aux` and define a new function `add_val` to `with_top nat`? This would have more functorial properties.

Reply:
I think adding `_aux` to the name is a useful flag that this is meant as implementation detail. You could also explicitly state that in the docstring.